base/util.go

```diff
@@ -1435,3 +1435,15 @@ func IsConnectionRefusedError(err error) bool {
 
 	return strings.Contains(err.Error(), errorMessage)
 }
+
+// JSONDecoderWithOptions returns a new JSON decoder implementing the JSONDecoderI interface with the specified options.
+func JSONDecoderWithOptions(r io.Reader, disallowUnknownFields, useNumber bool) (d JSONDecoderI) {
```

Comment:
Since we're never calling this with anything other than disallowUnknownFields=true and useNumber=true, I don't think I see the value in it.  It also makes it more difficult to see at a glance what the calling code is doing, I think - see comment below.

Reply:
Make sense, I've removed this and updated the comments below.